napari_tiff/napari_tiff_reader.py

```diff
@@ -299,7 +321,86 @@ def imagej_reader(tif):
         blending=blending,
         visible=visible,
     )
-    return [(data, kwargs, 'image')]
+    return kwargs
+
+
+def get_ome_tiff(tif, data):
+    layer_data = []
+    metadata = xml2dict(tif.ome_metadata)
+    if 'OME' in metadata:
+        metadata = metadata['OME']
+
+    series = tif.series[0]
+    shape = series.shape
+    dtype = series.dtype
+    axes = series.axes.lower().replace('s', 'c')
+    if 'c' in axes:
+        channel_axis = axes.index('c')
+        nchannels = shape[channel_axis]
+    else:
+        channel_axis = None
+        nchannels = 1
+
+    image = ensure_list(metadata.get('Image', {}))[0]
+    pixels = image.get('Pixels', {})
+
+    pixel_size = []
+    size = float(pixels.get('PhysicalSizeX', 0))
+    if size > 0:
+        pixel_size.append(get_value_units_micrometer(size, pixels.get('PhysicalSizeXUnit')))
+    size = float(pixels.get('PhysicalSizeY', 0))
+    if size > 0:
+        pixel_size.append(get_value_units_micrometer(size, pixels.get('PhysicalSizeYUnit')))
+
+    channels = ensure_list(pixels.get('Channel', []))
+    if len(channels) > nchannels:
+        nchannels = len(channels)
+
+    is_rgb = (series.keyframe.photometric == PHOTOMETRIC.RGB and nchannels in (3, 4))
+
+    scale = None
+    if pixel_size:
+        scale = pixel_size
+
+    for channeli, channel in enumerate(channels):
+        if not is_rgb and channel_axis is not None:
+            # extract channel data
+            if isinstance(data, list):
+                data1 = [numpy.take(level_data, indices=channeli, axis=channel_axis) for level_data in data]
+            else:
+                data1 = numpy.take(data, indices=channeli, axis=channel_axis)
+        else:
+            data1 = data
+        name = channel.get('Name')
+        color = channel.get('Color')
+        colormap = None
+        if color:
+            colormap = int_to_rgba(int(color))
+        elif is_rgb and len(channels) > 1:
+            # separate channels provided for RGB (with missing color)
+            colormap = ['red', 'green', 'blue', 'alpha'][channeli]
+            if not name:
+                name = colormap
+
+        contrast_limit = None
+        if dtype.kind != 'f':
+            info = numpy.iinfo(dtype)
+            contrast_limit = (info.min, info.max)
+
+        blending = 'additive'
+        visible = True
+
+        meta = dict(
```

Comment:
I understand the desire for consistency, but kwargs is the default Python name for keyword arguments. Why give this dictionary a non-meaningful name (agreeing with your comment above ;) ) only because eventually this will be passed down into a kwargs variable somewhere.
I used 'meta' as this corresponds with the definition of Layer.create() https://napari.org/stable/api/napari.layers.Layer.html#napari.layers.Layer.create

Reply:
I don't mean to be enforcing on this, just that naming is indeed important. If you really like kwargs we can do that too.